prod-aws/kafka-shared-msk/TERRAFORM_DEBUG.md

```diff
@@ -0,0 +1,32 @@
+## How to work with Terraform
+This guide is intended for **unexpected** situations, such as fixing the state or debugging the Kafka provider by the PubSub administrators.
+
+Under normal circumstances, we utilize the [Terraform Applier](https://github.com/utilitywarehouse/terraform-applier) to apply changes upon merge.
+
+## Introduction
+
+We have a pre-configured setup for this Terraform project, connecting to the MSK cluster within the PubSub namespace. This setup is represented by the stateful set `tf-kafka-config-admin`, available in both [dev](https://github.com/utilitywarehouse/kubernetes-manifests/tree/master/dev-aws/pubsub/tf-kafka-config-admin) and [prod](https://github.com/utilitywarehouse/kubernetes-manifests/tree/master/prod-aws/pubsub/tf-kafka-config-admin).
+
+This setup was initially created for manually deleting topics in `prod`, but it can also be employed for debugging the Terraform modules as needed.
+
+However, it’s essential to note that this setup always checks out the `main` branch of the project using git-sync as a sidecar container under the `uwgh` folder. For more details, see [manifest](https://github.com/utilitywarehouse/dev-enablement-mono/blob/1f026eb42591444aef2e9ae644d0c7b78c25a210/services/tf-kafka-config-admin/k8s-manifests/statefulset.yaml#L61-L78).
+
+### Steps to Initiate a Debug Session
+
+1. **Start a shell session** in the pod:
+    ```bash  
+    kubectl exec -n pubsub --context=dev-aws -it tf-kafka-config-admin-0 -- /bin/sh 
+    ```
+2. **Checkout this project in /tmp** (it must be anywhere else than `/uwgh` to not interfere with git-sync)
+    ```bash  
+    cd /tmp
+    git clone https://github.com/utilitywarehouse/kafka-cluster-config.git
+    ```
+3. **Initialise the terraform module** you're interested in:
+    ```bash
+    source /scripts/tf.sh
+    export REPO_ROOT=/tmp/kafka-cluster-config/
+    tfinit the_module
```

Comment:
Could include a `cd` here so someone can just copy paste the commands and things should just work :tm: 

```suggestion
    cd /tmp/kafka-cluster-config
    source /scripts/tf.sh
    REPO_ROOT="$PWD" tfinit the_module
```

Reply:
They are copy/pasteble as is, as I tested them. No need for cd, as it will be done by tfinit once REPO_ROOT is set. Is good to be exported too, so that tfinit could be invoked multiple times